src/components/viewers/MultiFlatmap.vue

```diff
@@ -52,6 +52,34 @@ const checkMarkersAtZoomLevel = (flatmapImp, markers, zoomLevel) => {
   }
 }
 
+const extractS3BucketName = uri => {
+  if (uri) {
+    const substring = uri.split("//")[1]
+    if (substring) {
+      return substring.split("/")[0]
+    }
+  }
+  return undefined
+}
+
+const getBodyScaffold = async(sparcApi, species) => {
+  //Get body scaffold information
+  const response = await fetch(`${sparcApi}get_body_scaffold_info/${species}`);
+  if (response.ok) {
+    const data = await response.json();
+    //Construct the url endpoint for downloading the scaffold
+    const bucket = extractS3BucketName(data.s3uri);
+    return `${sparcApi}s3-resource/${data.id}/${data.version}/files/${data.path}?s3BucketName=${bucket}`;
+  } else {
+    //Use default url if data is not found for any reason
+    if (species === "rat") {
+      return "https://mapcore-bucket1.s3.us-west-2.amazonaws.com/WholeBody/31-May-2021/ratBody/ratBody_syncmap_metadata.json";
```

Comment:
The other species can't make a request for a whole body scaffold right?

Reply:
No, we have decided to do that for rat and human. The rat dataset does not work well ( facing the opposite directly) so it is a hardcoded path for now.